subsys/net/l2/ieee802154/ieee802154_radio_aloha.c

```diff
@@ -49,6 +50,11 @@ static enum net_verdict aloha_radio_handle_ack(struct net_if *iface,
 {
 	struct ieee802154_context *ctx = net_if_l2_data(iface);
 
+	if (IS_ENABLED(CONFIG_NET_L2_IEEE802154_RADIO_CSMA_CA) &&
+	    ieee802154_get_hw_capabilities(iface) & IEEE802154_HW_CSMA) {
```

Comment:
question: Is handling ack not needed here?

Reply:
Current implementation of 802.15.4 MAC in Zephyr uses ACK only for software CSMA/CA, so no, it's not needed. Actually it's faulty in case of HW CSMA/CA, as kernel primitives used in `handle_ack` are only initialized in case HW CSMA/CA is disabled. It's explained in the commit message.